### ISSUE TITLE
InvalidCastException calling CreateFolderV2Async & folder is also created

### ISSUE THREAD

**Issue by AlirezaMaddah** [dropbox/dropbox-sdk-dotnet#193]
Hello, 
1. We are using the following code to create a folder:
```
        public async Task<string> CreateFolderAsync(string memberId, string newPath)
        {
            var args = new CreateFolderArg(path: newPath, autorename: false);
            var response = await _client.AsMember(memberId).Files.CreateFolderV2Async(args).ConfigureAwait(false);
            return response.Metadata.Id;
        }
```
2. We call the above method to create a folder with the iso-formatted datetime string:

`var folderId = await CreateFolderAsync("<adminMemberId>", "/Folder/SubFolder/2020-12-16T00:14:27.000Z");`

3. Expected Behavior: The folder should be created without any exception:
4. Actual Behavior: The folder is created and the following exception is thrown at the same time:

Message: System.InvalidCastException: 'Value '12/16/2020 00:14:27' is not valid System.String type

Do you confirm this issue as a bug?

Dropbox.Api: version 5.5.0
Visual Studio: 2019 v16.8.3

**Comment by greg-db** [comment]
Thanks for the report! Yes, this appears to be a bug in the SDK. I'll ask the team to fix this up.

**Comment by AlirezaMaddah** [comment]
Thanks, Greg.
When do you plan to fix it? 
We've noticed the same behavior in the following SDK methods as well:
1. DeleteV2Async: While deleting a folder with a "DateTime" format. The folder is deleted, but the above exception is also thrown.
2. ListFolderAsync: If the folder contains any subfolder with the DateTime format, it throws the above exception.

Thanks!

**Comment by greg-db** [comment]
Thanks for the additional information. It does appear this occurs with those other methods as well, as they use the same metadata types. This is open with engineering, but I can't offer a timeline for a fix.

**Comment by connorworley** [comment]
Hi, the bugfix should now be available in v6.0.1.
